CMakeLists.txt

```diff
@@ -23,6 +23,14 @@ if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
   message(FATAL_ERROR "In-source builds are not allowed.")
 endif ()
 
+# Check whether remnants of an earlier in-source build exist.
+if (EXISTS "${CMAKE_SOURCE_DIR}/CMakeCache.txt")
```

Comment:
You could also check for `CMakeFiles` since you mention that in the error message.

Reply:
Added a check for this.